tests/unit/Core/Usecase/ReceiveMessageTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * Integration test for ReceiveMessage usecase
+ */
+
+namespace Tests\Unit\Core\Usecase;
+
+use Ushahidi\Core\Usecase\ReceiveMessage;
+
+use Ushahidi\App\Listener\CreatePostFromMessage;
+use Ushahidi\App\Listener\HandleTargetedSurveyResponse;
+use Ushahidi\Core\Entity\Contact;
+use Ushahidi\Core\Entity\ContactRepository;
+use Ushahidi\Core\Entity\Message;
+use Ushahidi\Core\Entity\MessageRepository;
+use Ushahidi\Core\Entity\Post;
+use Ushahidi\Core\Entity\PostRepository;
+use Ushahidi\Core\Entity\TargetedSurveyState;
+use Ushahidi\Core\Entity\TargetedSurveyStateRepository;
+use Ushahidi\Core\Entity\FormAttribute;
+use Ushahidi\Core\Entity\FormAttributeRepository;
+
+use Tests\TestCase;
+use Mockery as M;
+
+/**
+ * @backupGlobals disabled
+ * @preserveGlobalState disabled
+ */
+class ReceiveMessageTest extends TestCase
```

Comment:
This test is a bit weird but I really wanted something that tested the ReceiveMessage usecase *with* the dispatcher and the listeners to verify I actually wired it up right. 
It turned out I didn't so I fixed that too :)